### PR TITLE
get okta system logs into graylog

### PR DESCRIPTION
**Pull request by ceruleancee**
design a proxy rest endpoint within graylog to obtain okta syslogs, so that we can execute the get request in swagger

- [x] Account activation with super admin rights.
- [x] generate Okta API Tokens. 
- [x] validate account, token using curl,postman and graylog [http json input](https://docs.graylog.org/en/3.2/pages/sending_data.html#json-path-from-http-api-input)
- [x] [protect your sensitive data. ](https://developer.okta.com/blog/2017/08/23/five-java-tips#5-protect-your-sensitive-bits)  
- [ ] Headers () 
- [x] [Request Parameters](https://developer.okta.com/docs/reference/api/system-log/#request-parameters)
- [ ] [LogEventObject](https://developer.okta.com/docs/reference/api/system-log/#logevent-object-annotated-example) 
- [ ] error handling. Ref :[Okta errors](https://developer.okta.com/docs/reference/error-codes/)
- [ ] design dynamic url's for polling - paginate using `next` parameter in the `link` that is returned in the response. Ref: https://github.com/Graylog2/graylog-plugin-integrations/issues/420
- [ ] unit tests.   
- [ ] [syslog queries usecase](https://support.okta.com/help/s/article/Useful-System-Log-Queries) 
- [ ]  reports.
- [ ]  default fields  in the graylog search page  [actor, target, time ]
- [ ] get the okta sdk for java up and running https://github.com/okta/okta-sdk-java#list-system-logs
- [ ] caching in single vs multi node jvm cluster 

**Technical Reference:**

-  IndexSetsResource.java ` in the Graylog ecosystem.
-  AWSKinesisResource`
-  [Rest](https://www.ics.uci.edu/~fielding/pubs/dissertation/top.htm)
-  https://developer.marvel.com/docs#!/public/getCreatorCollection_get_0
-  https://square.github.io/okhttp/

### CODE REVIEW

File: src/main/java/org/graylog/integrations/okta/OktaService.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    public OktaResponse getSystemLogs(String domain, String apiKey) throws IOException {
+        // TODO check if client is suitable
+        OkHttpClient client = new OkHttpClient().newBuilder().build();
+        Request request = new Request.Builder()
+                .url(domain)
+                .method("GET", null)
+                .addHeader("Accept", "application/json")
+                .addHeader("Content-Type", "application/json")
+                .addHeader("Authorization", apiKey)
+                .build();
```
Comment (danotorrey) [inline src/main/java/org/graylog/integrations/okta/OktaService.java]: This `OkHttpClient` is good to use here. 👍 However, please use the `OkHttpClientProvider` to obtain an instance through Guice dependency injection (some global options are set on it, and this pattern allows them to be applied for your usage). Here's an [example](https://github.com/Graylog2/graylog-plugin-enterprise/blob/15d44edb3dc0ec5b3a03b60734882ab310337886/enterprise/src/main/java/org/graylog/plugins/license/LicenseReportPeriodical.java#L78)

File: src/main/java/org/graylog/integrations/okta/OktaService.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import org.graylog2.shared.inputs.MessageInputFactory;
+
+import java.io.IOException;
+
+public class OktaService {
+
+    private final InputService inputService;
+    private final MessageInputFactory messageInputFactory;
+    private final NodeId nodeId;
+    private final ObjectMapper objectMapper;
```
Comment (danotorrey) [inline src/main/java/org/graylog/integrations/okta/OktaService.java]: Some unused fields.

File: src/main/java/org/graylog/integrations/okta/OktaResource.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Api(value = "Okta", description = "Okta Integrations")
+@Path("/okta")
+@Produces(MediaType.APPLICATION_JSON)
+public class OktaResource extends AbstractInputsResource implements PluginRestResource {
```
Comment (danotorrey) [inline src/main/java/org/graylog/integrations/okta/OktaResource.java]: I am not up-to-speed on the requirements, so just a question regarding that: Are both the Okta Input and the API endpoint needed?

File: src/main/java/org/graylog/integrations/okta/OktaService.java (hunk 1 above)
Comment (lingpri) [reply src/main/java/org/graylog/integrations/okta/OktaService.java]: Thanks @danotorrey for the pointer. also using this link as reference.
[okhttp](https://square.github.io/okhttp/)

File: src/main/java/org/graylog/integrations/okta/OktaResource.java (hunk 3 above)
Comment (lingpri) [reply src/main/java/org/graylog/integrations/okta/OktaResource.java]: yes. API endpoint for okta is to reach okta syslogs thru graylog API browser.  Okta Input is for Http polling.

**Comment by ceruleancee** [comment]
another issue has been created https://github.com/Graylog2/graylog-plugin-enterprise-integrations/issues/125

**Pull request closed without merging**